AddedCode

calnix opened the pull request:
Added my actual code

calnix committed:
AddedCode

calnix committed:
Update Readme.me

alcueca commented on src/SimpleNameRegister.sol:
```diff
@@ -0,0 +1,26 @@
+contract SimpleNameRegister {
+    
+    // map a string to an address to identify current owner
+    mapping (string => address) public nameOwner;    
```
```suggestion
    mapping (string => address) public holder;    
```
I don't want to use `owner` because people might think it refers to the contract owner. Other than that, I think we can be succint.

calnix replied:
Understood. Changes made.

alcueca commented on src/SimpleNameRegister.sol:
```diff
@@ -0,0 +1,26 @@
+    event NameRelinquished(address indexed owner, string indexed name);
+
+    // register an available name
+    function registerName(string memory _name) public {
```
```suggestion
    function register(string memory _name) public {
```
You could even consider `hold` as the function names, since we talk about `holders`. Please change the events accordingly.

calnix replied:
Understood. Changes made to both function name and events.

alcueca commented on src/SimpleNameRegister.sol:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    //owner can relinquish a name that they own
+    function relinquishName(string memory _name) public {
```
```suggestion
    function release(string memory _name) public {
```

alcueca commented on src/SimpleNameRegister.sol:
```diff
@@ -0,0 +1,26 @@
+    event NameRelinquished(address indexed owner, string indexed name);
+
+    // register an available name
+    function registerName(string memory _name) public {
```
```suggestion
    function registerName(string memory name) public {
```
We don't need underscores in every parameter. We just add one at the end in constructors and setters if there is a collision with a state variable.

calnix replied:
Understood. Removed underscore from function parameters.

alcueca commented on src/SimpleNameRegister.sol:
```diff
@@ -0,0 +1,26 @@
+
+    // register an available name
+    function registerName(string memory _name) public {
+        require(nameOwner[_name] == address(0), "The provided name has already been registered!");
```
```suggestion
        require(nameOwner[_name] == address(0), "Already registered!");
```
`require` strings should be 32 bytes or less, otherwise they use 2 words in memory.

calnix replied:
Thanks, changes made.

alcueca commented on test/SimpleNameRegister.t.sol:
```diff
@@ -0,0 +1,60 @@
+
+    // user can register an available name
+    function testRegisterName(string memory _testString) public {
+        simpleNameRegister.registerName(_testString);
```
Please ask Richie about building state in the tests using inheritance.

alcueca reviewed (changes requested):
The assignment is correct, with minor requests:
 - Shorter variable and function names.
 - Shorter require strings
 - State building in tests

alcueca commented on src/SimpleNameRegister.sol:
```diff
@@ -0,0 +1,26 @@
+
+    // emit event when name is registered or relinquished
+    event NameRegistered(address indexed owner, string indexed name);
+    event NameRelinquished(address indexed owner, string indexed name);
```
I'm not sure that you can index strings in events, would you check?

calnix replied:
Fundamentally, it can be done.

Indexed event arguments are stored in topics, and must be of fixed size; Solidity hashes the string and stored it in the topic. Filtering by the hash value of string would work. 

However, hash functions are one-way. We cannot decode the hashes of a specific address to get its list of names. Hence, I decided to not index string in my revised code. 

Thanks for pointing this out, learnt something new about events.

reference: 
https://ethereum.stackexchange.com/questions/6840/indexed-event-with-string-not-getting-logged

calnix committed:
Update src/SimpleNameRegister.sol

mapping nameOwner -> holder

Co-authored-by: Alberto Cuesta Cañada <38806121+alcueca@users.noreply.github.com>

calnix committed:
Update SimpleNameRegister.sol

1) mapping renamed to 'holder'
2) functions renamed to register and release; events updated to match
3) removed underscores from function parameters
4) require error strings shortened

calnix committed:
Modified require statements 

Renamed mapping in the require statements to match new naming convention (holder)

calnix committed:
Create compareTest

calnix committed:
Delete compareTest.t.sol
